packages/ingest/test/test-consumerRateLimited.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const sinon = require('sinon');
+const test = require('ava');
+
+const { sleep } = require('@cumulus/common');
+const SQS = require('@cumulus/aws-client/SQS');
+const SFN = require('@cumulus/aws-client/StepFunctions')
+
+const { ConsumerRateLimited } = require('../consumerRateLimited');
+
+const timeToReceiveMessages = 200; // ms
+const sqsMessage = { Body: 'message', MessageId: 'id' };
+const fakeQueueName = 'test-queue-url';
+let testConsumer;
+
+async function stubReceiveSQSMessages(_url, { numOfMessages }) {
+  await sleep(timeToReceiveMessages);
+  // eslint-disable-next-line prefer-spread
+  return Array.apply(undefined, { length: numOfMessages }).map(() => sqsMessage);
+}
+
+sinon.stub(SQS, 'deleteSQSMessage').resolves();
+sinon.stub(SQS, 'receiveSQSMessages').callsFake(stubReceiveSQSMessages);
+
+function processFn() { }
+
+const sandbox = sinon.createSandbox();
+let messageSpy;
+
+test.beforeEach(() => {
+  // need to reinstantiate because this.now = Date.now()
+  testConsumer = new ConsumerRateLimited({});
```

Comment:
This works, functionally, because we're using .js tests, but in prod this would fail as the constructor sets these - should we set the class internals via the constructor in case that method changes and *doesn't* break this test? 

Also is .messageLimit valid here?   Etc.

Reply:
Will move this to the new PR if it comes up in the code there.